v2/logger/logger.go

```diff
@@ -0,0 +1,9 @@
+package logger
```

Comment:
This is always tough :)

I noticed that many things in this lib have callbacks like OnError, OnUpdate etc... I'm curious to hear thoughts around the inclusion of logging bits vs. strict exclusion and letting the user bring their own logger in combination with the callbacks

Reply:
I think you want both. Logging for diagnostics, callbacks for reactivity. In order to make logging meaningful in a callback-only world, you'd need a lot of granular callbacks to for each internal step, and those callbacks would need to be passed a bunch of (sometimes internal) details in order to make the logs actionable. All of those callbacks would complicate the API. It would also have the potential of backcompat hell if we decided to provide more details to a particular callback. You'd also get folks taking weird dependencies on callbacks for other unrelated reasons that might also further complicate backcompat as the library grew.